Tests/AsyncHTTPClientTests/HTTPClientInternalTests.swift

```diff
@@ -403,37 +408,68 @@ class HTTPClientInternalTests: XCTestCase {
 
         // cannot test with NIOTS as `maxMessagesPerRead` is not supported
         let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
-        let httpClient = HTTPClient(eventLoopGroupProvider: .shared(eventLoopGroup))
-        let delegate = BackpressureTestDelegate(eventLoop: httpClient.eventLoopGroup.next())
+        let eventLoop = eventLoopGroup.next()
+        let writeBodyPromise = eventLoop.makePromise(of: Void.self)
+        let writeEndPromise = eventLoop.makePromise(of: Void.self)
         let httpBin = HTTPBin { _ in
             WriteAfterFutureSucceedsHandler(
-                bodyFuture: delegate.optionsApplied.futureResult,
-                endFuture: delegate.backpressurePromise.futureResult
+                bodyFuture: writeBodyPromise.futureResult,
+                endFuture: writeEndPromise.futureResult
             )
         }
-
         defer {
-            XCTAssertNoThrow(try httpClient.syncShutdown(requiresCleanClose: true))
             XCTAssertNoThrow(try httpBin.shutdown())
             XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully())
         }
 
         let request = try Request(url: "http://localhost:\(httpBin.port)/custom")
-
-        let requestFuture = httpClient.execute(request: request, delegate: delegate).futureResult
+        let logger = Logger(label: "test-connection")
+
+        let clientFactory = HTTPConnectionPool.ConnectionFactory(
```

Comment:
Why has this dropped down to be so low-level?

Reply:
I would really like to not expose the connection on the task in the future. In order to have control over the channel, we need to create it ourselves here.

Reply:
There is only one other test remaining that accesses the `task.connection`:

`testWeCanActuallyExactlySetTheEventLoops`

We should be able to remove `task.connection` from `testWeCanActuallyExactlySetTheEventLoops`, once the new connection pool has landed. We would test only the delegate calls, since all connection eventLoop requirements can be tested with unit tests and will be tested with unit tests.

Reply:
Ok, can we encapsulate this new code in a function that clearly explains what it does? The code here is _very_ aware of implementation details of AHC, which suggests the test as a whole needs rewriting, but in the near term I'd be ok with just hiding the implementation details in a function.

Reply:
This is still open.

Reply:
Bump. :smile:

Reply:
Yes, ignored it for now, since I think we should be able to just remove this test, once other PRs have landed. The functionality will be tested by unit tests.